<?xml version="1.0" encoding="utf-8"?>	

<launch>

  <arg name="robot_name" default="ugv1" />
  <arg name="simulator" default="/vrep" />
  
  <arg name="respawn_value" default="false" /> <!-- boolean: true, false -->
  
  <arg name="lambda_trav"  default="1.0"/>    

    <node name="queue_path_planner_$(arg robot_name)" pkg="path_planner" type="queue_path_planner" 
        respawn="$(arg respawn_value)" output="screen">

	      <param name = "lambda_trav" value="$(arg lambda_trav)"/>  
        <param name = "robot_frame_name" value = "$(arg robot_name)/base_link"/>

        <remap from = "/clustered_pcl/wall" to = "$(arg simulator)/$(arg robot_name)/clustered_pcl/wall"/>
        <remap from = "/trav/traversability" to = "$(arg simulator)/$(arg robot_name)/trav/traversability"/> 
        
        <remap from = "/path_planner/visited_nodes" to = "$(arg simulator)/$(arg robot_name)/path_planner/visited_nodes"/>
        <remap from = "/path_planner/localPath" to = "$(arg simulator)/$(arg robot_name)/path_planner/localPath"/>

        <remap from = "/planner/tasks/path" to = "$(arg simulator)/$(arg robot_name)/planner/tasks/path"/>
        <remap from = "/planner/tasks/global_path" to = "$(arg simulator)/$(arg robot_name)/planner/tasks/global_path"/>
        <remap from = "/planner/tasks/feedback" to = "$(arg simulator)/$(arg robot_name)/planner/tasks/feedback"/>
        <remap from = "/planner/tasks/append" to = "$(arg simulator)/$(arg robot_name)/planner/tasks/append"/>
        <remap from = "/planner/tasks/remove" to = "$(arg simulator)/$(arg robot_name)/planner/tasks/remove"/> 
        <remap from = "/planner/waypoints/cropbox" to = "$(arg simulator)/$(arg robot_name)/planner/waypoints/cropbox"/>      
    
    </node> 
  
</launch>
